rust/lance/src/io/exec/planner.rs

```diff
@@ -514,6 +538,38 @@ impl Planner {
             }
         })
     }
+
+    /// Determine how to apply a provided filter
+    ///
+    /// We parse the filter into a logical expression.  We then
+    /// split the logical expression into a portion that can be
+    /// satisfied by an index search (of one or more indices) and
+    /// a refine portion that must be applied after the index search
+    pub fn create_filter_plan(
+        &self,
+        filter: &str,
```

Comment:
It'd be nice that we can bump pyarrow version and support strait from it.

Reply:
What do you mean?